p2p/host/basic/basic_host.go

```diff
@@ -143,17 +150,30 @@ func NewHost(net inet.Network, opts *HostOpts) *BasicHost {
 		h.cmgr = opts.ConnManager
 	}
 
+	var relayCtx context.Context
+	var relayCancel func()
+	if opts.Relay {
+		relayCtx, relayCancel = context.WithCancel(context.Background())
```

Comment:
I think this context is off -- it's not wired to anything, but is used to dial peers, open streams, and open relayed streams. It should be wired to *something* so it correctly gets cancelled when shutting down go-ipfs.